module/zfs/noload.c

```diff
@@ -95,29 +155,29 @@ static void bio_map_buf(struct bio *bio, void *data, unsigned int len)
 		if (len <= 0)
 			break;
 
-		if (bytes > len)
-			bytes = len;
+		if (bytes > len && !IS_ALIGNED(len, ALGO_ALIGN))
+			return bio_bounce_pad(bio, data, len, is_dst);
 
 		if (!is_vmalloc)
 			page = virt_to_page(data);
 		else
 			page = vmalloc_to_page(data);
 
-		bio_add_page(bio, page, bytes, offset);
```

Comment:
If bytes > len then it shouldn't get here.... So why min(bytes, len)?

Reply:
The condition is `bytes > len && !IS_ALIGNED(len, ALGO_ALIGN)`